src/server/main_service.cc

```diff
@@ -2161,6 +2161,18 @@ void Service::PubsubPatterns(ConnectionContext* cntx) {
   (*cntx)->SendLong(pattern_count);
 }
 
+void Service::PubsubNumSub(const vector<string_view>& channels, ConnectionContext* cntx) {
+  vector<std::pair<std::string, long>> pubsub_num_array;
```

Comment:
`NUMSUB` shouldn't be on the hot path, so it's fine if it's not fully optimized.
However, if you wish to make it slightly more efficient, you could avoid the creation of this `vector` and instead call:
`(*cntx)->StartArray(channels.size() * 2);`
And then for each `channel` simply do: `SendBuildString(...); SendLong(...);`

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/0abca50381ec31ee7332cc95617af1ec9dad1eae